src/common.cpp

```diff
@@ -1997,6 +1997,25 @@ void assert_is_locked(void *vmutex, const char *who, const char *caller) {
     }
 }
 
+/// Detect if we are Windows Subsystem for Linux by inspecting /proc/sys/kernel/osrelease
+/// and checking if "Microsoft" is in the first line.
+/// See https://github.com/Microsoft/WSL/issues/423
+bool is_windows_subsystem_for_linux() {
+    ASSERT_IS_NOT_FORKED_CHILD();
```

Comment:
Do we need this here? From experience trying to clean up the fork code, these asserts can be misleading.

Reply:
Can you say more about how they're misleading? The problem I'm trying to address here is that this function is not safe to call after fork, because it can allocate memory. The assert hopes to be both documentation and a runtime check.